### DISCUSSION TITLE
Lesson 6: Error when compiling MockV3Aggregator- "TypeError: Interfaces cannot inherit"

### DISCUSSION THREAD

**Discussion by strudelPie** [smartcontractkit/full-blockchain-solidity-course-py#875] [Q&A] [answered]
I've copied the MockV3Aggregator.sol code as shown but when I try to compile I get hit with the following error. Any ideas? 

```
CompilerError: solc returned the following errors:

C:/Users/[user]/.brownie/packages/smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol:7:38: TypeError: Interfaces cannot inherit.
interface AggregatorV2V3Interface is AggregatorInterface, AggregatorV3Interface
                                     ^-----------------^
C:/Users/[user]/.brownie/packages/smartcontractkit/chainlink-brownie-contracts@1.1.1/contracts/src/v0.6/interfaces/AggregatorV2V3Interface.sol:7:59: TypeError: Interfaces cannot inherit.
interface AggregatorV2V3Interface is AggregatorInterface, AggregatorV3Interface
```

**Comment by cromewar** [comment]
Hello @strudelPie  Interfaces are abstract contract and should not be inherit by other contracts, could you please share the whole code in order to check it?

**Reply by strudelPie** [reply]
that's what I thought! Thanks- but which bit as the code is just completely copied and pasted as per the tutorial from the link below (without it and up to this point no issues at all): 
[https://github.com/smartcontractkit/chainlink-mix/blob/master/contracts/test/MockV3Aggregator.sol](url)

**Comment by strudelPie** [comment, marked as answer]
Ok so I have sorted it out for me as follows, in case it helps anyone else. Still not quite sure why only I seem to be getting this issue(?)

Replaced the import statement with:

```
import "@chainlink/contracts/src/v0.6/interfaces/AggregatorInterface.sol";
import "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
```

and then changed the contract declaration to be:

`contract MockV3Aggregator is AggregatorInterface, AggregatorV3Interface`

Compiled fine after that.

**Reply by cromewar** [reply]
Nice hint, you are awesome !

**Reply by kevinszuchet** [reply]
You're awesome. Thanks for sharing your solution!

**Reply by lfsr** [reply]
Thanks! Could you hint us how did you arrive to the solution?
